dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -361,10 +364,21 @@ private DependencyGraphResult levelOrder(
           }
         }
       }
+
+      // If false, the resulting tree does not have duplicate artifacts with regard to
+      // groupId:artifactId. This behavior is an optimization for Linkage Checker.
+      boolean allowDuplicateNodeInTree =
+          graphTraversalOption != GraphTraversalOption.FULL_DEPENDENCY_WITH_PROVIDED;
```

Comment:
This seems error prone. In particular this converts a multivalue to a boolean. But it could easily be the case that there are multiple values that allow duplicate nodes in the tree and multiple other values that don't, if not now then in the future. 

Perhaps this would be more obvious if the different traversal algorithms were different methods instead of being configured by an option.